src/formatDate.js

```diff
@@ -50,7 +50,34 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldFormat = date.split(fromFormat[3]);
+  const newFormat = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (let j = 0; j < fromFormat.length - 1; j++) {
+      if (fromFormat[j].includes(toFormat[i].at(0))) {
+        if (toFormat[i].at(0) === 'Y') {
+          if (toFormat[i].length === 2) {
```

Comment:
I don't understand this code :( Lets rewrite it to more readable structure. Move every logical action to separate variable/function